doc/architecture/decisions/0015-web-container-interface.md

```diff
@@ -0,0 +1,98 @@
+- Define a protocol to use over the WebSocket so that clients can take advantage of container features such
+  as talking to multiple apps
+- Make use of sessions and a permission system to only expose certain apps or capabilities to remote UIs
+
+### 3. Daemon container communicating with browser plug-in via IPC
+This is similar to MetaMask for Ethereum communicating with a local node via IPC. 
+- Plug-in exposes Holochain calls to the JavaScript running on the page via an injected object
+- Plug-in communicates with daemon via an IPC file or similar
+
+Not sure yet what substantial advantage this holds over browser without plug-in.
+
+### 4. Holochain container runs a custom browser that exposes calls directly
+This is comparable to the Mist browser for Ethereum and currently implemented for QML
+front-ends with [HoloSqape](https://github.com/holochain/holosqape). 
+This could be re-done with web technology such as [Electron](https://electronjs.org/) and [Holochain-nodejs](https://github.com/holochain/holochain-nodejs).  
+- Similar to above but removes the need for IPC
+- Would need to continue running in the background when no windows are open
+
+
+### Other Discussions
+
+There is also the possibility to implement multiple of the above (as Ethereum has). This might be more practical where there are many more people working on core. The real problem is which to do first
+
+## Decision
+
+Option 2.
+
+The existing Holochain container project HoloSqape is to be extended to provide a **WebSocket interface**
+(next to the QML based GUI plug-in system) that enables many different kinds of external
+user interfaces to connect to the local Holochain node, authenticate, issue zome calls and receive zome signals,
+depending on permissions administered by the local user of the node through an admin UI or default settings.
+
+A **JavaScript based client library** is to be implemented that makes it easy to use this interface from the context
+of a web browser, hiding the complexity and wire-protocol of the websocket interface itself, only offering high-level
+functionality such as:
+* connecting to a local or remote node
+* authentication (public-key?)
+* quering installed/running apps, the session has access to
+* issuing zome function calls and receiving results
+* registering callbacks for zome signals 
+
+![](../WebSocket-interface-HoloSqape.png)
+
+## Consequences
+
+* WebSocket interface needs to be designed (upcoming ADR) and implemented
+* App developers have an alternative to QML based UIs
+* Web based UIs can be build and run separately from the Holochain node itself, both locally and remotely
+* Web based UIs can just be a browser tab since Holochain app keeps running in HoloSqape without it
+* UI and Holochain node are connected through permanent bi-directional connection
+* Push notifications can be send from the node to the external UI 
+* Connection between Holo light-client and HoloPort could be implemented on top of this
```

Comment:
That is what I would like to discuss in tomorrow's meeting as well. As I said in the screencast, I believe that we can build this interface in a general way so that it can help with many different use-cases and I would like to understand the requirements you mention, @neonphog.